tests/string.in

```diff
@@ -392,4 +392,24 @@ function dualsplit
 end
 count (dualsplit)
 
+logmsg string collect
+count (echo one\ntwo\nthree\nfour | string collect)
+count (echo one | string collect)
+echo [(echo one\ntwo\nthree | string collect)]
+echo [(echo one\ntwo\nthree | string collect -c)]
+printf '[%s]\n' (string collect one\n\n two\n)
+printf '[%s]\n' (string collect -c one\n\n two\n)
+# string collect always returns 0
```

Comment:
That's of course one way to go about it. It might be useful though if e.g. it returned 1 if there was nothing to collect?

So you could do

```fish
if set -l var (somecommand | string collect)
```

and you wouldn't have to add a `test -n` after?

Reply:
So basically, return `1` if `count (… | string collect)` outputs `0`? That sounds reasonable. I assume parsing stdin automatically handles this by just having zero arguments; I'll have to check what the `string collect ''` case does, since `count (string collect '')` outputs `0` even though I gave an explicit arg, but maybe the arg iterator handles that case too.

Reply:
Done